File: algorithms/2_medium/V/03969_valid_subarrays_with_matching_sum_digits_i/bruteforce.py
class Solution:
    def countValidSubarrays(self, nums: list[int], x: int) -> int:
        assert(x >= 1 and x <= 9)
        n = len(nums)
        cnt = 0

        for i in range(n):
            s = 0
            for j in range(i, n):
                s += nums[j]
                lowest_dig = s % 10
                if lowest_dig == x:
                    temp = s
                    while temp >= 10:
                        temp //= 10
                    highest_dig = temp
                    if highest_dig == x:
                        cnt += 1

        return cnt

File: algorithms/2_medium/V/03969_valid_subarrays_with_matching_sum_digits_i/bruteforce.rs
impl Solution {
    pub fn count_valid_subarrays(nums: Vec<i32>, x: i32) -> i32 {
        assert!(x >= 1 && x <= 9);
        let x = x as i64;
        let len = nums.len();
        let mut cnt = 0;

        for i in 0..len {
            let mut sum = 0_i64;
            for j in i..len {
                sum += nums[j] as i64;
                let lowest_dig = sum % 10;
                if lowest_dig == x {
                    let mut temp = sum;
                    while temp >= 10 {
                        temp /= 10;
                    }
                    let highest_dig = temp;
                    if highest_dig == x {
                        cnt += 1;
                    }
                }
            }
        }

        cnt
    }
}
